src/app/legacy/containers/PageHandlers/withData/shouldRender/index.js

```diff
@@ -17,11 +26,17 @@ const shouldRender = (
       service,
       passportHomesOverride,
     );
-    statusCode = !isCorrectService ? 404 : status;
+    if (service === 'sport' && pageType === ARTICLE_PAGE) {
+      const canonicalUrl = getCanonicalUrl(pageData);
+      isCanonicalUrlMatch = matchesCanonicalUrl(canonicalUrl, pathName);
+    }
+    const isUrlValid = isCorrectService && isCanonicalUrlMatch;
+    statusCode = !isUrlValid ? 404 : status;
   }
 
   return {
-    hasData200StatusAndCorrectService: hasDataAnd200Status && isCorrectService,
+    hasData200StatusAndCorrectService:
+      hasDataAnd200Status && isCorrectService && isCanonicalUrlMatch,
```

Comment:
could `isUrlValid` be used in place of `isCorrectService && isCanonicalUrlMatch`?